lib/command/update/general_lastup_updater.rb

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2013 whiteleaf. All rights reserved.
+#
+
+require_relative "../../narou/api"
+
+module Command
+  class Update < CommandBase
+    class GeneralLastupUpdater
+      def initialize(options)
+        @options = options
+        @database = Database.instance
+        @narou_novels = Hash.new { |h, k| h[k] = [] }
+        @other_novels = []
+        prepare
+      end
+
+      def save
+        @database.save_database
+      end
+
+      def prepare
+        @database.each.with_index do |(id, data), i|
+          next if Narou.novel_frozen?(id)
+          setting =  Downloader.get_sitesetting_by_target(id)
+          if setting["narou_api_url"]
+            @narou_novels[setting["narou_api_url"]] << setting["ncode"]
+          else
+            @other_novels << id
+          end
+          setting.clear
+        end
+      end
+
+      def update_narou_novels
+        @narou_novels.each do |api_url, ncodes|
+          api = Narou::API.new(api_url: api_url, ncodes: ncodes, of: "nu-gl")
+          api.request.each do |result|
+            ncode = result["ncode"]
+            data = Downloader.get_data_by_target(ncode)
+            if result["novelupdated_at"] > data["last_update"]
+              data["novelupdated_at"] = result["novelupdated_at"]
+              data["general_lastup"] = result["general_lastup"]
+              tags = data["tags"] ||= []
+              tags << "modified" unless tags.include?("modified")
+            end
+          end
+          @other_novels += api.private_novels
+        end
+      end
+
+      def update_other_novels
+        progressbar = ProgressBar.new(@other_novels.size - 1)
+        interval = Interval.new(@options["interval"])
+
+        @other_novels.each_with_index do |id, index|
+          progressbar.output(index)
+          interval.wait
+          begin
+            setting =  Downloader.get_sitesetting_by_target(id)
+            info = NovelInfo.load(setting)
```

Comment:
__[rubocop]__
* Lint/UselessAssignment: Useless assignment to variable - `e`. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

<!-- Comment by [SideCI](https://sideci.com) -->